src/Stratis.Bitcoin/Utilities/Extensions/CollectionExtensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Stratis.Bitcoin.Utilities.Extensions
+{
+    /// <summary>
+    /// Extension methods for collections.
+    /// </summary>
+    [DebuggerStepThrough()]
+    public static class CollectionExtensions
+    {
+        /// <summary>
+        /// An extension that will check if an <see cref="IEnumerable{T}"/> is empty.
+        /// </summary>
+        /// <typeparam name="TSource">The type of enumerable.</typeparam>
+        /// <param name="source">The enumerable to check.</param>
+        /// <returns><o>True</o> if empty otherwise <o>False</o>.</returns>
+        public static bool IsEmpty<TSource>(this IEnumerable<TSource> source)
+        {
+            return !source.DefaultIfEmpty().Any();
+        }
+
+        /// <summary>
+        /// An extension that will check if an <see cref="IList{T}"/> is empty.
+        /// </summary>
+        /// <typeparam name="TSource">The type of enumerable.</typeparam>
+        /// <param name="source">The enumerable to check.</param>
+        /// <returns><o>True</o> if empty otherwise <o>False</o>.</returns>
+        public static bool IsEmpty<TSource>(this IList<TSource> source)
+        {
+            return (source == null) || (source.Count == 0);
```

Comment:
can you really do that with null? i mean if you have `List x = null;`
`x.IsEmpty()` would work?